api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImpl.java

```diff
@@ -110,6 +113,42 @@ public RelatedPerson toFhirResource(@Nonnull Relationship relationship) {
 	
 	@Override
 	public Relationship toOpenmrsType(@Nonnull RelatedPerson resource) {
-		throw new UnsupportedOperationException();
+		Relationship relationship = new Relationship();
```

Comment:
We can't always generate a new relationship... there needs to be some allowance for updating an existing relationship, right? So I think we need to change `RelatedPersonTranslator` to extend `UpdatableOpenmrsTranslator`. Likewise `personA` should be retrieved from the database, if at all possible... Honestly, I think that's the really hard part about this....